{
  if (exception == null) {
    future.complete(returnValue);
  }
 else {
    future.completeExceptionally(exception);
  }
}
